compaction/compaction_manager.cc

```diff
@@ -994,19 +994,12 @@ class compaction_manager::offstrategy_compaction_task : public compaction_manage
             }
         });
 
-        for (;;) {
+        auto get_next_job = [&] {
             auto& iop = service::get_local_streaming_priority(); // run reshape in maintenance mode
-            auto desc = t.get_compaction_strategy().get_reshaping_job(reshape_candidates, t.schema(), iop, sstables::reshape_mode::strict);
-            if (desc.sstables.empty()) {
-                // at this moment reshape_candidates contains a set of sstables ready for integration into main set
-                auto desc = sstables::compaction_completion_desc{
-                    .old_sstables = std::move(old_sstables),
-                    .new_sstables = std::move(reshape_candidates)
-                };
-                co_await t.update_sstable_lists_on_off_strategy_completion(std::move(desc));
-                break;
-            }
+            return t.get_compaction_strategy().get_reshaping_job(reshape_candidates, t.schema(), iop, sstables::reshape_mode::strict);
+        };
```

Comment:
I guess you can make it return std::optional<> to avoid the operator bool injected into compaction_descriptor.